### PR TITLE
Update notebooks - Hybrid Scan and Incremental Refresh

### LINKED ISSUE #246: [FEATURE REQUEST]: Update notebooks to showcase hybrid scan scenarios for all languages
### Describe the issue

Currently Hitchhiker's docs on hyperspace only showcase basic usages of index maintenance and usage apis. It doesn't include an example of hybrid scan with append and delete scenarios. We need to add e2e examples of hybrid scan on all three languages.

### LINKED ISSUE #247: [FEATURE REQUEST]: Update notebooks to showcase Refresh Index with incremental mode scenarios for all languages
### Describe the issue

Currently Hitchhiker's docs on hyperspace only showcase basic usages of index maintenance and usage apis. It doesn't include an example of refresh incremental scenarios. We need to add e2e examples of refresh incremental for all three languages.

### PR DESCRIPTION
**Pull request by sezruby**
### What is the context for this pull request?

 - **Tracking Issue**: #246, #247 
 - **Parent Issue**: n/a
 - **Dependencies**: n/a

Fixes #246
Fixes #247

### What changes were proposed in this pull request?

Updated notebooks for Hybrid Scan and Incremental Refresh.

### Does this PR introduce _any_ user-facing change?

Yes, updated notebooks for Scala/Python/C#

### How was this patch tested?

Scala / Python - HDInsight 2.4 Jupyter Notebook
C# - Azure Synapse Notebook

### CODE REVIEW

**Comment by sezruby** [comment]
@reviewers, should we add cells for importing hyperspace?
```
%%configure -f
{ "conf": {"spark.jars.packages": "com.microsoft.hyperspace:hyperspace-core_2.11:0.3.0" }}
```

File: notebooks/csharp/Hitchhikers Guide to Hyperspace.ipynb [hunk 1]
```diff
@@ -12,10 +12,14 @@
       "cell_type": "markdown",
       "metadata": {},
       "source": [
-        "## Hitchhiker's Guide to Hyperspace - An Indexing Subsystem for Apache Spark™\n",
-        "Hyperspace introduces the ability for Apache Spark™ users to create indexes on their datasets (e.g., CSV, JSON, Parquet etc.) and leverage them for potential query and workload acceleration.\n",
+        "# Hitchhiker's Guide to Hyperspace (.NET for Spark C#)\n",
```
Comment (pirz) [inline notebooks/csharp/Hitchhikers Guide to Hyperspace.ipynb:15]: Should not it be "C# for Spark .Net"?

File: notebooks/csharp/Hitchhikers Guide to Hyperspace.ipynb [hunk 2]
```diff
@@ -210,6 +211,8 @@
         "* An Apache Spark™ DataFrame which references the data to be indexed.\n",
         "* An index configuration object: IndexConfig, which specifies the *index name*, *indexed* and *included* columns of the index. \n",
         "\n",
+        "As you might have noticed, in this notebook, we illustrate indexing using the [Covering Index](https://www.red-gate.com/simple-talk/sql/learn-sql-server/using-covering-indexes-to-improve-query-performance/), which are the default index in Hyperspace. In the future, we plan on adding support for other index types. \n",
```
Comment (pirz) [inline notebooks/csharp/Hitchhikers Guide to Hyperspace.ipynb:214]: "... which is the default index in Hyperspace".

**Review by pirz — APPROVED** [review]
LGTM, except below two nit comments. Thanks @sezruby

File: notebooks/csharp/Hitchhikers Guide to Hyperspace.ipynb (hunk 1 above)
Comment (sezruby) [reply notebooks/csharp/Hitchhikers Guide to Hyperspace.ipynb:15]: Fixed.

**Comment by apoorvedave1** [comment]
LGTM, thanks @sezruby

**Review by imback82 — APPROVED** [review]

**Pull request merged**